### PR TITLE
[RF] POM AT > 09.01_006 | Create new view

### PR DESCRIPTION
**Pull request by SvetlanaVet**
https://trello.com/c/9AkfnoPU/1804-rf-pom-at-0901006-my-viewscreate-new-view
[RF] POM AT > 09.01_006 | <My Views>Create new view

### CODE REVIEW

**Comment by Forostinko** [comment]
You need to change the name of the describe. Ut's not clear what kind of check you are doing

**Comment by 7108581** [comment]
Your PR should be named like the RF card is named

**Review by ElenaKanash — APPROVED** [review]
There are a few comments:
1. in the it() must be the name of the AT card 
 2. test name contains only the creation of the project, but in the test - checking the name. it would be more correct to add checking the name to the test name
 3.  creation of the project can be moved to a separate method in commands.js (it has already been written in this file)

**Review by ArinaJur — APPROVED** [review]

File: cypress/e2e/tests/myNewView.cy.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import HomePage from "../../pageObjects/HomePage"
+import newItemPage from "../../fixtures/pom_fixtures/newItemPage.json"
+import myView from "../../fixtures/pom_fixtures/myView.json"
```
Comment (Rom-T) [inline cypress/e2e/tests/myNewView.cy.js:3]: Please rename to `myViewData` and `newItemPageData`

File: cypress/e2e/tests/myNewView.cy.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        .clickAddNewViewRadioBtn()
+        .clickDescriprionSaveBtn()
+        .getNameMyView().should('have.text', myView.viewName)
+    })
+
+    after('delete view', () => {
+        myViewPage
+        .clickDeleteViewBtn()
+        .clickMyViewDeleteOkBtn();
+    })
```
Comment (Rom-T) [inline cypress/e2e/tests/myNewView.cy.js:32]: Why is this removal necessary? Before each test, we automatically delete views.

File: cypress/e2e/tests/myNewView.cy.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    const myViewPage = new MyViewPage(); 
+      
+    it('My Views Create new view', () => {
+        homePage
+        .clickCreateJobLink()
+        .typeNewItemNameInputField(newItemPage.freestyleProjectName)
+        .selectFreestyleProjectItem()
+        .clickOkBtnAndGoFreestyleProjectConfig()
+        .clickSaveBtnAndGoFreestyleProject()
+        .clickHomePageLink()
```
Comment (Rom-T) [inline cypress/e2e/tests/myNewView.cy.js:19]: These steps are not needed since the creation of the project has already been moved to commands. You can use the command `cy.createFreestyleProject(newItemPageData.freestyleProjectName);` at the beginning of the test. And then start with HomePage.

File: cypress/pageObjects/MyViewPage.js [hunk 4]
```diff
@@ -160,6 +164,31 @@ clickMultiBranchPipelineNameLink(){
     return this;
   };
 
+  clickNewViewBtn() {
+    this.getAddNewViewLink().click()
+    return this;
+  }
```
Comment (Rom-T) [inline cypress/pageObjects/MyViewPage.js:170]: Instead, we need to use the verifyAndClickAddNewViewLink() method as we navigate to the NewViewPage.

File: cypress/pageObjects/MyViewPage.js [hunk 5]
```diff
@@ -160,6 +164,31 @@ clickMultiBranchPipelineNameLink(){
...
+
+  clickAddNewViewRadioBtn() {
+    this.getAddNewViewRadioBtn().click();
+    return this;
+  }
+
+  clickDescriprionSaveBtn() {
+    this.getDescriprionSaveBtn().click();
+    return this;
+  }
```
Comment (Rom-T) [inline cypress/pageObjects/MyViewPage.js:185]: All these methods and elements are on NewViewPage.

File: cypress/e2e/tests/myNewView.cy.js (hunk 1 above)
Comment (SvetlanaVet) [reply cypress/e2e/tests/myNewView.cy.js:3]: Sorry, I forgot about it. Done

File: cypress/e2e/tests/myNewView.cy.js (hunk 3 above)
Comment (SvetlanaVet) [reply cypress/e2e/tests/myNewView.cy.js:19]: Thanks
